service/state/height.go

```diff
@@ -32,20 +31,9 @@ type Height struct {
 // => https://github.com/awfm9/flow-dps/issues/37
 
 func (h *Height) ForBlock(blockID flow.Identifier) (uint64, error) {
-	key := make([]byte, 1+len(blockID))
-	key[0] = prefixIndexBlock
-	copy(key[1:], blockID[:])
 	var height uint64
 	err := h.core.db.View(func(tx *badger.Txn) error {
-		item, err := tx.Get(key)
-		if err != nil {
-			return fmt.Errorf("could not retrieve block index: %w", err)
-		}
-		_ = item.Value(func(val []byte) error {
-			height = binary.BigEndian.Uint64(val)
-			return nil
-		})
-		return nil
+		return RetrieveBlockHeight(Encode(prefixIndexBlock, blockID[:]), &height)(tx)
```

Comment:
Following previous comments, this would become `RetrieveHeightByBlock(blockID, &height)(tx)`.